vmdb/app/controllers/ems_infra_controller.rb

```diff
@@ -33,41 +33,56 @@ def scaling
       return
     end
 
-    @count_parameters = @stack.parameters.select { |x| x.name.include?('::count') }
+    @orig_parameters = @stack.parameters.select { |x| x.name.include?('::count') }
 
     return unless params[:scale]
 
-    scale_parameters = params.select { |k, _v| k.include?('::count') }
-    assigned_hosts = scale_parameters.values.sum(&:to_i)
+    form_parameters = params.select { |k, _v| k.include?('::count') }
+
+    validate_selected_hosts_not_exceed_available_hosts(form_parameters)
+
+    changed_parameters = []
```

Comment:
hm there should be no need to pass this to the method, rather return it from select_changed_parameters_and_return_message